common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -126,4 +150,38 @@ private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
       }
     }
   }
+
+  private class DownloadCallback implements StreamCallback {
+
+    private WritableByteChannel channel = null;
+    private File targetFile = null;
+    private int chunkIndex;
+
+    public DownloadCallback(File targetFile, int chunkIndex) throws IOException {
+      this.targetFile = targetFile;
+      this.channel = Channels.newChannel(new FileOutputStream(targetFile));
```

Comment:
@zsxwing @cloud-fan 
`OneForOneBlockFetcher` "open blocks" asynchronously. If I understand correctly, the retry of the `start()` in `OneForOneBlockFetcher` is only triggered when failure of sending `OpenBlocks`, but failure of fetching chunk cannot trigger the retry in `RetryingBlockFetcher`. `DownloadCalback` is not initialized when the failure of "open blocks" happens. So there cannot be two `DownloadCallbacks` for same stream working at the same time.